src/main/java/io/github/railroad/project/ui/BrowseButton.java

```diff
@@ -19,6 +20,35 @@ public class BrowseButton extends Button {
     private final ObjectProperty<BrowseSelectionMode> selectionMode = new SimpleObjectProperty<>(BrowseSelectionMode.SINGLE);
     private final ObjectProperty<Path> defaultLocation = new SimpleObjectProperty<>(Path.of(System.getProperty("user.home")));
 
+    public DirectoryChooser FolderBrowser(File defaultPath, String title){
+        DirectoryChooser directoryChooser = new DirectoryChooser();
+        directoryChooser.setTitle(title);
+        directoryChooser.setInitialDirectory(defaultPath);
+
+        return directoryChooser;
+    }
+
+    public FileChooser FileBrowser(File defaultPath, String title, @Nullable FileChooser.ExtensionFilter filter){
+        FileChooser fileChooser = new FileChooser();
+        fileChooser.setTitle(title);
+        fileChooser.setInitialDirectory(defaultPath);
+        if(filter != null){
+            fileChooser.setSelectedExtensionFilter(filter);
+        }
+
+        return fileChooser;
+    }
+
+    public FileChooser ImageBrowser(File defaultPath, String title){
+        FileChooser.ExtensionFilter filter = new FileChooser.ExtensionFilter("Image Files", "*.png", "*.jpg", "*.jpeg", "*.gif");
+
+        FileChooser fileChooser = new FileChooser();
+        fileChooser.setTitle(title);
+        fileChooser.setInitialDirectory(defaultPath);
+        fileChooser.setSelectedExtensionFilter(filter);
+
+        return fileChooser;
+    }
```

Comment:
`camelCase` for methods. Also, make these `static` since they do not require an instance.